File: Sources/PylorsAssetMgr/Models/TagSchema.swift
import Foundation

enum TagSource: String, CaseIterable, Codable, Sendable {
    case system
    case user
}

struct TagColumnInfo: Identifiable, Sendable {
    let key: String
    let source: TagSource
    var defaultVisible: Bool = false
    var defaultWidth: Int = 120
    var humanName: String

    var id: String { key }

    init(key: String, source: TagSource, defaultVisible: Bool = false,
         defaultWidth: Int = 120, humanName: String = "") {
        self.key = key
        self.source = source
        self.defaultVisible = defaultVisible
        self.defaultWidth = defaultWidth
        self.humanName = humanName.isEmpty ? key : humanName
    }
}

File: pylors_asset_mgr/models/tag_schema.py
from dataclasses import dataclass
from enum import Enum, auto


class TagSource(Enum):
    SYSTEM = auto()   # 系统标签，实时从文件系统读取
    USER = auto()     # 用户标签，SQLite 存储


@dataclass
class TagColumnInfo:
    key: str
    source: TagSource
    default_visible: bool = False
    default_width: int = 120
    human_name: str = ""

    def __post_init__(self):
        if not self.human_name:
            self.human_name = self.key
